src/cromshell/list_outputs/command.py

```diff
@@ -0,0 +1,200 @@
+                )
+            else:
+                print_file_like_value_in_dict(
+                    outputs_metadata=get_workflow_level_outputs(config).get("outputs"),
+                    indent=False,
+                )
+        else:
+            if json_summary:
+                io_utils.pretty_print_json(format_json=get_task_level_outputs(config))
+            else:
+                print_task_level_outputs(get_task_level_outputs(config))
+
+    return return_code
+
+
+def get_workflow_level_outputs(config) -> dict:
+    """Get the workflow level outputs from the workflow outputs
+
+    Args:
+        config (dict): The cromshell config object
+    """
+
+    requests_out = requests.get(
+        f"{config.cromwell_api_workflow_id}/outputs",
+        timeout=config.requests_connect_timeout,
+        verify=config.requests_verify_certs,
+        headers=http_utils.generate_headers(config),
+    )
+
+    if requests_out.ok:
+        return requests_out.json()
+    else:
+        http_utils.check_http_request_status_code(
+            short_error_message="Failed to retrieve workflow outputs.",
+            response=requests_out,
+            # Raising exception is set false to allow
+            # command to retrieve outputs of remaining workflows.
+            raise_exception=False,
+        )
+
+
+def get_task_level_outputs(config) -> dict:
+    """Get the task level outputs from the workflow metadata
+
+    Args:
+        config (dict): The cromshell config object
+    """
+    # Get metadata
+    formatted_metadata_parameter = metadata_command.format_metadata_params(
+        list_of_keys=config.METADATA_KEYS_TO_OMIT,
+        exclude_keys=True,
+        expand_subworkflows=True,
+    )
+
+    workflow_metadata = metadata_command.get_workflow_metadata(
+        meta_params=formatted_metadata_parameter,
+        api_workflow_id=config.cromwell_api_workflow_id,
+        timeout=config.requests_connect_timeout,
+        verify_certs=config.requests_verify_certs,
+        headers=http_utils.generate_headers(config),
+    )
+
+    return filer_outputs_from_workflow_metadata(workflow_metadata)
+
+
+def filer_outputs_from_workflow_metadata(workflow_metadata: dict) -> dict:
+    """Get the outputs from the workflow metadata
+
+    Args:
+        workflow_metadata (dict): The workflow metadata
+    """
+    calls_metadata = workflow_metadata["calls"]
+    output_metadata = {}
+    extract_task_key = "outputs"
+
+    for call, index_list in calls_metadata.items():
+        if "subWorkflowMetadata" in calls_metadata[call][0]:
+            output_metadata[call] = []
+            for scatter in calls_metadata[call]:
+                output_metadata[call].append(
+                    filer_outputs_from_workflow_metadata(scatter["subWorkflowMetadata"])
+                )
+        else:
+            output_metadata[call] = []
+            for index in index_list:
+                output_metadata[call].append(index.get(extract_task_key))
```

Comment:
what if a task has no output? In theory it is possible. Then the `get` call will return `None`. Does the downstream handle `None` correctly? I think it's better if we handle that here.

Reply:
good catch, done